### DISCUSSION TITLE
[Update: Reimplemented in Cypress 10.7] Cypress Studio’s removal in Cypress 10.0

### DISCUSSION THREAD

**Discussion by pstakoun** [cypress-io/cypress#21561] [General]
**Update as of August 30, 2022: Cypress Studio has been reimplemented in Cypress 10 as of version 10.7.0! Please see https://github.com/cypress-io/cypress/discussions/21561#discussioncomment-3511688 for more info.**

We’re excited to share that Cypress 10.0 will soon be released. It’s packed with new features, like first-class support for Component Testing.

Cypress Studio, the experimental feature first introduced in 6.3.0, will be removed in Cypress 10.0. For more information about why, check out this [blog post](https://cypress.io/blog/2022/05/18/update-on-cypress-studio/).

As we assess what we learned about Studio during its experimental period, we’d like to hear your feedback.

- What does Studio mean to you and your organization?
- What would make Studio more valuable?
- Did Studio solve any challenges you faced with other test generation solutions?

Please share your thoughts below. Your feedback will help us prioritize features to make writing Cypress tests even easier.

**Edit (17th Aug 2022)**: We are actively working on this. See: https://github.com/cypress-io/cypress/discussions/21561#discussioncomment-3410059

**Edit (30th Aug 2022)**: We have [reimplemented](https://github.com/cypress-io/cypress/discussions/21561?sort=new#discussioncomment-3511688) Cypress Studio into Cypress 10 as of version 10.7.0, which was released today! Please see our [blog post](https://www.cypress.io/blog/2022/08/30/how-to-use-studio-in-cypress-10-7) for more info.

**Comment by sarala02** [comment]
This is feature is very useful as no coding knowledge  required also very easy to add assertions as well ,I would recommend to keep this feature going forward .

**Comment by howardpanton** [comment]
This is a useful feature for creating spec files, we use it daily at our organisation

**Comment by tmbarbour** [comment]
Our team really likes the integrated Cypress Studio recorder in Cypress Test Runner. The ability to generate tests right in the Cypress UI, the ease of recording and adding steps to existing tests and then for it to run immediately and use Cypress Playground to fine tune the test are used extensively. To be able to create a single test offline, save it to a file, run a command to convert and then import it to your project is at least a hack, but a big vote to get it integrated back into Cypress in the future.

**Comment by bjan99** [comment]
I really like the integrated Cypress Studio recorder. I use it every day. I find it very helpful to make a quick first pass at creating my test. Then I easily clean it up a bit. I also use it to make a quick addition to an already created test. Saves me a lot of time. I have also used it when showing other people Cypress for the first time. They love it and it gets them interested in Cypress. I looked at the Cypress plugin for DevTools Recorder, but it is not as easy as the integrated Cypress Studio. Please bring it back.

**Comment by jonwest3** [comment]
I think Studio is very valuable. We switched from Selenium to Cypress because of the ease of use and integrated Cypress Studio was a big part of that. My team and I give a big vote also to get it integrated back into Cypress. Glad I just heard about it's removal. I was pitching we get Cypress Dashboard, but will now hold off on that until I see if it get's integrated back in soon.

**Comment by michelnicholas** [comment]
Studio is very valuable I like all the features currently with Cypress, it really helps me and my team understand why and where exactly scenarios are failing. Its is highly valuable please don't not remove if anything don't remove just add. If you remove i probably will not update lol.

**Comment by dwelsh1** [comment]
I have used Cypress studio to get many of my hesistant manual testers to take the jump into automation.

While recording test scenarios won't create a 100% complete test script, it is an easy way to start one.  When they see that work, then they are excited and want to learn more to be able to add/change to make the script great.

With tools like Selector Playground and Cypress Studio, the Cypress team has opened the door to automation to everyone.

The Cypress plugin/Devtool recorder is not a good replacement compared to the integrated Studio.

Yes, Studio is very valuable especially to those just starting out in automation, please bring it back.

**Comment by ViktorijaFilipov** [comment]
My team is also heavily using Cypress Studio. We find it an amazing tool for people that are just getting into automation. As previous comments said - selector playground, debugging, and other features really make our lives easier. Please support it in the future, we need it.

**Comment by SaraCooperQA** [comment]
I think Cypress Studio is very valuable and should not be removed. And this is why I think so.

I was the only QA and a strictly manual tester with no programming experience on my team. We had no automated tests and the team wanted me to look into starting writing them. A developer mentioned Selenium. So I started down a path with Selenium with Java. After 3 days and a lot of help getting everything set up to run on my laptop and a lot of help writing my first automated smoke test, it was done but every now and then would fail.

During a stand-up meeting I said I was struggling and another developer mentioned I might want to try Cypress. He heard it was pretty east to get started with. I found the cypress site and read the nice Getting Started section. I got it installed without needing any help, so off to a great start.

I wanted to see how my Cypress journey compared to my Selenium journey so was keeping track of time. I wanted to see how long it would take me to write the same test in Cypress. After just under 2 hours I had the test completed and had ran it several dozen times and no strange failures every now and then. The 2 hours included setup, some cypress documentation reading and some time googling/researching. All this with not having to bother the developers! It was about 30 minutes of actual test writing. Much better than 3 days with Selenium.

During my research I came across a great article by Filip Hric called Getting Started with Cypress Studio for Test Automation https://applitools.com/blog/getting-started-cypress-studio/. As an experiment, I wanted to see how long it would take to create the same smoke test using Cypress Studio. It took less than 5 minutes to generate the test. I did some cleanup to it and added a few assertions that I couldn’t see a way to add from Studio. For example there is a dropdown that I wanted to assert has 3 items in it. 

All together I spent about 10 minutes cleaning and adding to the Studio generated test for a total of 15 minutes. So 15 minutes total for Cypress Studio vs 30 minutes writing it myself. A great time savings! Your mileage may vary depending on many factors such as your overall automation and programming experience, the length / complexity of the test you are writing, the app your testing etc.

So for me, I love Cypress Studio and still use it every day several months later. I think if Studio was promoted more by Cypress, it would be used by more people. It would have been nice to have a Using Cypress Studio in the Getting Started section of the Cypress docs. The only reason I knew about Studio at all was by luck coming across the article by Filip. 

With Cypress version 10 coming out, I was hoping Studio would no longer be an experimental feature and maybe would have some small improvements, but then saw the very sad news it was being removed.

Sorry for the long post and thanks for reading if you made it this far!

**Comment by SaraCooperQA** [comment]
Some improvements for Studio could be having some options (since not everyone may want them) like:

- Not including the Generated by Cypress Studio comments.
- Chaining commands like if you have a clear line and then a type text line just chain them.
- Options to have a single cy and no ;
- More assertions like a count for a dropdown.

A small example of a generated test and then a cleaned up version. It would be nice to have some options to get the cleaned up one without doing the cleanup manually each time:

  /* ==== Test Created with Cypress Studio ==== */
  it('clear-completed', function() {
    /* ==== Generated with Cypress Studio ==== */
    cy.visit('localhost:8000');
    cy.get('.new-todo').clear();
    cy.get('.new-todo').type('Clear Me{enter}');
    cy.get('.toggle').check();
    cy.get('.clear-completed').click();
    cy.get('.new-todo').should('have.attr', 'placeholder', 'What needs to be done?');
    /* ==== End Cypress Studio ==== */
  });
  
  **After cleanup:**

    it('clear-completed', function() {
    cy
    .visit('localhost:8000')
    .get('.new-todo').clear().type('Clear Me{enter}')
    .get('.toggle').check()
    .get('.clear-completed').click()
    .get('.new-todo').should('have.attr', 'placeholder', 'What needs to be done?')
  });

**Comment by smcintosh881** [comment]
Im bummed to hear about cypress studio being dropped. Not only did it make writing tests incredibly quick, I see a lot of potential for non engineering job roles to be able to leverage the tools. PMs/QA/Management could easily help document these flows.

As a tool it also fits nicely into BDT development style

**Comment by peans99** [comment]
This is a really bad news. We were looking at acquiring license but without this feature we will have to rethink what we will do and look at alternative

**Reply by peans99** [reply]
no only that the google recorder don't work well with our react dynamic front end so its really not an option right now. that bring back selenium forward unless an alternative is borned even if its a paid option

**Comment by JoshuaGoddard** [comment]
I use the studio daily to quickly grab tag ID's and get the basics of the flow started. There are work arounds, but nothing works quite as well as the integrated studio. Sad to see it go.

**Comment by drdreo** [comment]
What does Studio mean to you and your organization? - it was a nice DX feature
What would make Studio more valuable? - finding the right selector is hard, our HTML structure is rather complex and we always had to polish those.
Did Studio solve any challenges you faced with other test generation solutions? - we never used any but manual generation

**Comment by Pachat** [comment]
The Studio is so necessary and useful that I instantly revert to 9.7.0 
And will not move from it until it is back in new Cypress versions.

Otherwise I'll seek some other solution than Cypress.

**Reply by luqy** [reply]
How to restore to 9.7.0

**Reply by Pachat** [reply]
It was easy in my case. I just had

- to reinstall >`npm i cypress@9.7.0 --save`
- verify version >`npx cypress -v`

and in case you let cypress 10 to do the renamings
- to rename back `cypress\e2e` to `cypress\integration`
- to rename back `cypress\support\e2e.js` to `cypress\support\index.js`

**Reply by luqy** [reply]
I tried to uninstall and reinstall it, but found that it was not completely uninstalled, and there were abnormal errors
I'll try the second method

**Reply by Pachat** [reply]
It was not necessary to uninstall
Just run `npm i cypress@9.7.0 --save` in the same directory you run the `npm i cypress@10.x.x --save`
and then, do the renaimings of `e2e`

**Reply by isolderea** [reply]
You also need to revert your support file. Do not forget, I forgot yesterday

**Comment by luqy** [comment]
This function is very good and can quickly produce scripts

**Comment by mirobo** [comment]
I could never use this function for real because as most recorders (even the new one built in into Google Chrome) it fails to "do the right thing" when the AUT has complex / unusual UI components. I have built an abstraction layer over most of these UI components and unless the recorder itself can be extended to make use of the abstraction layer (or at least incooporate custom commands) then I don't see a huge benefit other than get a rough outline of a test very quickly.

**Reply by Pachat** [reply]
In my case, I use it on complex UI components, and that's true that I have to refactor a few `cy.get` for more accurate and stable finds.
But, all over, it gives a useful scaffold to work with and makes the test ready much quickly.

**Comment by isolderea** [comment]
It was a really good tool to learn how to work and think in the Cypress world.
I used it sometimes as a backup solution when I could not find the correct way to interact with an element. I even recommended it to my friends and youtube subscribers. 

Please bring it back. The alternative that you suggested via the Chrome Recorder is not the same. It is not as user friendly, it is harder to use and learn.

**Comment by MNouroozi** [comment]
Why remove This feature? is very good and can quickly produce scripts...
i think you did that for money ...

**Comment by rodrigosouza83** [comment]
The feature Run all specs, support us in when we need to run a  test suite in a large scale.
This feature shouldn't  be removed.

**Comment by orchkarim** [comment]
Functionality is a good tool is one of the best arguments I can use for the team to change from Katalon to Cypress and we need to run a complex test suite.
This feature should not be removed please.

**Comment by rnunezil** [comment]
I was evaluating Cypress 9.x and the recorder (Studio) is the one of the main reasons I pitched Cypress to my team and now I find out it is removed in 10.  A lot of automation tools out there have a built in recorder.

**Comment by donwelsh** [comment]
I think Cypress Studio should be brought back as it is very valuable as shown by all the posts above. Cypress Studio was one of the reasons I also pitched Cypress to my organization. It allows manual testers to easily start working with automation. It also allows BA's and PM's to help easily create test scenarios. Then more experienced engineers can fine tune those tests.

The alternative that you suggested with the Chrome Recorder is not as good or as easy to use as the integrated Studio was.

Please bring Studio back.

**Comment by Wilhop** [comment]
Cypress Studio was a key factor into getting non-technical people to record tests. Especially the Assertions are something that no other Cypress-recorder can solve for Cypress to my knowledge.

Example: let's imagine software in a highly regulated industry. If the subject matter expert can define what flows are important and assert values on the way, technical people can focus on technical aspects only. Cypress Studio was the main reason for not going for TestProject.io.

**Comment by plfgavilan** [comment]
Please, bring it back. Studio was one of the best features of cypress. It is really a key feature.
It is a really really usefull feature.
You can just simple record new tests, and then check the code to clean it up.

For me it was a key feature, actually, I'm not doing the upgrade from the 9 version until you bring it back again the cypress studio (I'm pretty sure you will).

As an improvement for studio, I was expecting an improved selectors finder. It would be nice to have multiple options when the selector is not a cy-data etc.

**Comment by istateside** [comment]
Just to add a dissenting opinion here - I liked the idea of Cypress Studio, but it was never useful for my team. Our application is complicated, and involved stepping in and out of the context of an iframe. I can see it being useful for simple flows, but Studio was never really more than a "hey that's pretty cool" thing for the many engineers on my team working with Cypress.

I'm not suggesting it was actively bad or useless or anything like that, clearly there are people who got value from it. But there were certainly other things I would like to see come to Cypress before investing more time into Studio.

**Comment by mirshodtairov** [comment]
Studio is very helpful for me to bypass fats and easy validation which saves me lots of time during test development. Please bring it back asap!

**Comment by sultanzarif** [comment]
Cypress Studio is a great tool, it is unfortunate that it will be removed. Cypress should certainly keep it going forward!

**Comment by GarethCake** [comment]
Cypress Studio is an awesome tool, one that we use very frequently at our company! Please bring back Cypress Studio #studioftw.

EDIT: I also have reverted to Cypress 9 until I see Cypress Studio's return as we are using a RIOT frontend which doesn't appear to be supported by the new component testing anyway. Upgrade to a slightly more modern UI but lose Cypress Studio? I DONT THINK SO!

**Comment by stevenirby** [comment]
I too would like to add another voice to keeping this around.

**Comment by ghost** [comment]
I would love to add more to that, our company worked with other things, as soon as I introduced the studio it changed the attitude, lowered the fear of entering the world of testing and especially cypress, all QA people do tests, and thus learn to do more complex things. Instead of removing it, I would improve it further and further and further.
Even as an experienced programmer it is easier to start a test like this and then improve it. This way we will be able to create something of value really easily

**Comment by kkhanhluu** [comment]
This feature is very useful for tester and QAs in my team, who have no coding knowledge. Please bring this feature back

**Comment by dwelsh1** [comment]
@pstakoun Any word if and when Cypress Studio will return?

**Comment by victorcmarinho** [comment]
This feature is very useful for tester and QAs in my team, the way it generates the code is very useful for optimizing the work and making automations in less time.

Very sad that this feature has been removed, I hope it comes back in the future

**Comment by patrick-mota** [comment]
I feel a bit sad for Cypress team.

So many switched to Cypress because of the recorder and saw a bright future with Cypress team going into user direction.

It's really a key feature and that was the only feature that made us to switch to Cypress.

It's ok, you tried to bring something via the chrome recorder but a lot of plugins already exists in chrome which does better than this one (with asserts support) so I don't understand why you choose something with less features instead of improving an existing module which is already almost perfect.

What does Studio mean to you and your organization?
Without Studio or any tool which allow us to record easily with assert support we will reconsider Cypress.

What would make Studio more valuable?
A dedicated IDE or a dedicated module (chrome extension ?).

Did Studio solve any challenges you faced with other test generation solutions?
Recording was a lot better than the ones we found (Selenium recorder) for javascript/ajax pages.

**Comment by pstakoun** [comment]
Thank you all for your continued feedback. This thread has been super helpful for understanding the usage of the experimental Cypress Studio in the wild. I want to add that we're actively reviewing this feedback and are still working towards enabling everyone to write Cypress tests quickly and effectively.

Right now, our main question for current users of Studio is: have you tried other solutions that generate Cypress tests in interactive ways? For example, the Cypress Chrome Recorder (https://github.com/cypress-io/cypress-chrome-recorder) and Preflight for Cypress (https://cypress.preflight.com).

If so, are there any use cases that Studio addresses that these other solutions are unable to, and are there any use cases that none of the existing solutions (including Studio) address? If you have not tried these other solutions, are there any blockers to doing so that do not exist with Studio?

**Reply by kkhanhluu** [reply]
@pstakoun thanks for your information and your team's effort. I already tried the cypress chrome recorder and IMHO, there 2 drawbacks,: 
1. Assertion is not yet supported. 
2. I had to go back and forth between browser and terminal, download the json file, run the cli to generate the test and add assertions by myself. The whole testing process requires more effort and is much more complicated for testers and QAs, who have little (or no) coding knowledge. 

I haven't heard about the preflight for cypress, it look promising and i'll definitely try it, but I wonder if there's any limitation if I only use the free plan.

**Reply by peans99** [reply]
Thanks for the update.

We tried both and they really didn't work well with Dynamic React. Pre-flight couldn't get past the first page and even bypassing some of the dynamic it just not as mature of a recorder at this point. It fail to record a lot of the steps.

Google recorder doesn't position submit and was a hit and miss with dynamic doms. There is no assertion supported for it. 

They are both unable to compete with something like let say Katalon.

All of this is working with cypress studio in 9.X. So we are left without a path forward at this time with cypress.

**Reply by Wilhop** [reply]
Thumbs up for listening to the community and participating in the conversation (whatever the decision).
I tested out all the Cypress recorder options available and none of the free ones had Asserting functionalities.
-Chrome recorder itself is OK, but the process of importing the tests is just bothersome. But have to admit, maybe we could automate some stuff on that.
-Preflight has a simple Assertion, but for Free it's 10 recordings. 300$ for 50 recordings. I don't see an outcome where we would pay more for just a recorder than for an entire Test framework. It will be cheaper to create our own recorder.

**Reply by SaraCooperQA** [reply]
We tried both of the mentioned other recorders and they really didn't work well in our case (failed to record steps) and were not as nice and easy as the free integrated Cypress Studio.

As mentioned by others above, the Chrome recorder is not easy to use and has no assertions. Preflight costs money so is a no go for us anyways.

Thanks for listening.

**Reply by Samyssmile** [reply]
This is like, "Yea let us kill our main killer feature".

#insta revert to version 9

**Reply by dwelsh1** [reply]
We tried both of the mentioned other recorders as well. Like others have said they really didn't work well for us and were not as easy as the free integrated Cypress Studio. Please bring it back.

**Reply by carwei** [reply]
> -Preflight has a simple Assertion, but for Free it's 10 recordings. 300$ for 50 recordings. I don't see an outcome where we would pay more for just a recorder than for an entire Test framework. It will be cheaper to create our own recorder.

Did they Preflight change their pricing because of Cypress Studio being removed? Now only 5 tests are free and for 300 USD per month you only get 20(!) Cypress recordings. Sure, you get e-mail and visual testing credits included but I just want was Cypress Studio covered.

**Reply by patrick-mota** [reply]
> I want to add that we're actively reviewing this feedback and are still working towards enabling everyone to write Cypress tests quickly and effectively.

Can we have an ETA ?
It's been a month since your last comment, we need to move forward in our company so if you aren't planning to do anything and will just reply all the time "try this or this or we are waiting more feedback" it's a waste of time to us.

I took a look in the thread (took me around 2 hours) to gather information and 89% of the time people are just asking for the EXACT same feature as they had so you already know what people want.

I don't want to rush you guys, I know it's summer and things will slow down but if we could at least have an ETA like "we are planning to implement this feature back in october" or whenever you want, it could help at least to say to customers/company, ok we can't upgrade right now because of this drawback but don't worry, things will come back later this year !

Or an ETA saying, we don't plan anything, eventualy we will consider to put back this feature, but I think if you do this probably all will switch to another product because I'm not sure people was in hurry to get testing components introduced with your new version.

We are already using jest (we didn't knew cypress at that time) but since we saw cypress with studio it was the right choice for e2e.
If later we consider to replace jest it's only if cypress still the best for e2e testing and adds new features but otherwise we will just replace cypress with anything else for e2e.

**Reply by pstakoun** [reply]
Please see my update here: https://github.com/cypress-io/cypress/discussions/21561#discussioncomment-3158126

**Comment by carwei** [comment]
Cypress Studio was the only reason we decided on Cypress as our first go at automated testing. It's so easy to use to create a new test and iterate on it. I don't feel Cypress Studio had any big drawbacks in the experimental version. Our pain point has been with Cypress itself, mainly flakiness due to rerendering in React (which we think we will minimize over time). But, Cypress Studio is the reason we decided to stick with Cypress. Dropping it in version 10 will be really challenging for us. 

I looked at the alternatives mentioned, but as others have said it's a dealbreaker to be able to add assertions. The pricing for Preflight is impossible for us, we don't write an even amount of tests each month and the unlimited version is super expensive. Also, Studio being integrated with Cypress enabled a very streamlined workflow. We would definitely be willing to pay a reasonable flat fee to keep supporting Cypress Studio if that's what it takes. But Cypress + Studio is the package that brought us to Cypress.

**Comment by jonwest3** [comment]
My team and I tried the Chrome recorder and it was a pain to use compared to Cypress Studio. No one wants to use it. We looked at the website for Preflight but the pricing is a big no for us, so didn't bother to try it out.

In our opinion, nothing beats the nicely integrated Studio even in it's current experimental state (it was working fine with no issues for us). Yes, another +1 to bring Studio back!

**Comment by alapanme** [comment]
Cypress Studio was one of the easiest ways to learn and generate cypress scripts. A lot of PMs & Manual Testers in my team took interest in cypress because of this feature only, as it enabled them to do a quick hands-on. So my hope is Cypress Studio will come back bigger and better because it's a feature that provides a lot of value to end-users.

**Comment by jonwest3** [comment]
I wanted to mention that a couple of Cypress Ambassadors have written about Cypress Studio in the past and liked it.

Marie Drake: https://www.mariedrake.com/post/experimenting-with-cypress-studio

“I have been exploring [Cypress Studio] one of Cypress' experimental features and I have to say that this is currently one of my favourite features at the moment. It's perfectly suitable for beginners and to anyone who wants to quickly automate some tests with less or no coding at all.”

Filip Hric: https://applitools.com/blog/getting-started-cypress-studio/

“It is actually a great learning tool for all those brave souls that want to start with test automation. And that’s where Cypress Studio becomes really empowering.”

**Comment by LN-T** [comment]
Cypress Studio was a nice way to learn to write quick and simple tests, learning by doing is a great way to test a software. There are other plugins in Chrome but it was nicer and maybe safer to use the official one.

**Comment by msimkonis** [comment]
Our team has switched to this tool and found the recording feature very valuable. Please retain this feature as this has placed Cypress ahead of many other tools and their capabilities.

**Comment by Vfleitao** [comment]
I believe the studio is an essential feature for cypress.
It is the distinguishing feature against other, also javascript based frameworks. rather than removing it because it would be to much work, it should become one of the cornerstones of cypress since its easily the main benefit of this framework.

It allows during development that a non technical individual (ie: a manual qa) to get the tests created quickly using a visual interface and get them working repeatedly without much effort.
A more specialized user can then use it to convert it to a page model approach or whatever its needed for long term maintenance.

Without the studio, cypress loses most of its appeal for companies that are trying to adopt it to an existing product and even for a company with a fast paced development where a lot of times the tests need to be written or recorded quickly.

**Comment by SaraCooperQA** [comment]
Another reason Cypress Studio should be brought back...

Cypress Studio is very easy to use as it is integrated right into Cypress which allows quickly creating tests (including assertions) to get instant feedback. We started the development phase for a new product about a month ago. We quickly created a suite of Cypress tests using Studio. We now have confidence in the core features that have been developed so far that as more development occurs these critical features will not be broken. Since then I have tweaked those tests

Besides previous mentioned feature requests here are some more:
1. Add support for recording API tests and assert on the responses.
2. Add the use of variables. Let the user create and use dynamic variables. Such as a pattern for email addresses.
3. Add some default steps that can be added during recording such as Setting/removing cookies, Setting a viewport size, print a message to the Cypress Command Log etc.
4. Option for self healing selectors. Record different ways to select the element, then if the default way fails try the other ones.

**Comment by gkimpson** [comment]
I love Cypress Studio - this is one of the core reasons I use this tool over others - hopefully this returns soon

**Comment by szazadikablak** [comment]
Cypress Studio was the best option for newbies to learn automation. Please bring it back ASAP.

**Comment by elio-maggini** [comment]
Cypress Studio was the killer feature of Cypress, IMO.

Cypress enables a wide range of testing and is the best tool I have used.

But Studio made testing rather complex custom components built with Ionic and Angular very simple - a HUGE lift in test code generation.

So simple in fact that currently Cypress is the ONLY package.json entry that is not on "latest", it remains on 9.7.0.....which, I guess, you will have to pry from my cold, dead hands. :)

**Comment by M0BI0S** [comment]
This feature was the reason to have choosen cypress as a novice. Now installed, the feature has been removed. Please re-add this feature, it immensly helps to get started.

**Comment by lila1-work** [comment]
I'm really gutted this has been removed.

I'd been trying to coax a couple of team members off another unsupported version of product because the new one had dropped support for something they used, and they had come round to Cypress purely because of Studio as it was easy to record, copy/paste sections and add a screenshot later too.   (It would be a useful feature to add a screenshot in Studio itself though)

I have a couple of non-technical testers and they were using this without trouble, and it meant they could create scripts for the present time and I could help them refactor them later.

It was also nice that it was all in one tool.  Having a separate recorder makes it really faffy and makes hesitant testers even less confident in using it.

I'm loathe to remain on another unsupported version of a tool so I may have to find something else.

**Comment by pstakoun** [comment]
Hi all, I want to provide an update on the current state of Cypress Studio. Typically, we can prioritize feature work in one of two ways: we try to work on a bunch of different projects at the same time and move them all forward bit by bit, or we focus in on a handful and get them over the line and in the hands of users much faster. As of right now, we have made the decision to choose the latter and not continue work on Cypress Studio at this time. We are instead prioritizing the GA release of [cy.origin](https://docs.cypress.io/api/commands/origin) and [cy.session](https://docs.cypress.io/api/commands/session) (which are both currently in experimental), as well as beginning work for iFrame support ([#136](https://github.com/cypress-io/cypress/issues/136)) and resolving the common “element is detached from the DOM” error ([#7306](https://github.com/cypress-io/cypress/issues/7306)). This does not mean we have written off Studio indefinitely, but it does mean it will not see much movement over the next few months at the very least. Prioritization is always fluid, which is why we’re still actively monitoring this thread, the existing alternatives, and feedback from other areas. Thanks again for everyone’s feedback on this topic.

**Reply by peans99** [reply]
I hope the dashboard will remain compatible with version 9.7.0 for the next year as your decide on your priorities we have no intention of moving to 10 at this time.

**Reply by carwei** [reply]
Based on the feedback in this thread, the sensible thing would be to put Studio in the next handful of stuff to push after the current work. Dropping features (like Studio and the "run all tests in browser" feature) and having them pending indefinitely doesn't make sense for us as end-users.

**Reply by rustamd** [reply]
Watch them spin it around in couple months into "Look at this incredible enterprise only feature we just developed! Only $9999 a month!1!" (;

**Comment by Pachat** [comment]
@pstakoun Thank you for giving us information and hope.
It is not a problem if it takes months.
We will wait until the Studio is back to upgrade from 9.7.0

**Comment by pkrall520** [comment]
Is there a roadmap to add the recorder back, soon? If not then, we will have to downgrade to 9.7

**Reply by carwei** [reply]
They already said there is none, 3 days ago.

**Reply by lmiller1990** [reply]
I am looking into it: https://github.com/cypress-io/cypress/issues/22870, will see the logistics of reviving studio in Cy 10. It's not the highest priority, more focus is on cy origin and session, see https://github.com/cypress-io/cypress/discussions/21561#discussioncomment-3158126, but studio certainly is not forgotten or swept under the rug.

**Reply by dwelsh1** [reply]
Awesome news @lmiller1990 . Thanks for the update!

**Comment by jozzya** [comment]
It is a shame that Studio isn't going to be included with Cypress 10.

I think studio flattens the learning curve required to build "useful" cypress tests and get non JS developers involved with writing tests.
Eventually these non developers will find that cypress studio doesn't provide all of the functionality and they will have to learn the framework. However I feel that learning curve is not as big as the non JS would have been exposed to syntax.

I will downgrade from Cypress 10 down to 9.7 as non developer buy-in is more important in my use case.

**Comment by Capscreen** [comment]
Doesn't make that much sense in my opinion. One of the absolute benefits from your software is its simplicity and fast-pasted possibility to create tests. Aborting this core functionality is something that is probably not that beneficial for anyone. 
I understand technical challenges might seem hard but you definitely should work on that.

**Comment by elio-maggini** [comment]
No Studio support **AND** component testing doesn't support Angular? Where do I sign up?. 🤕

**Reply by pstakoun** [reply]
We are very actively working on Angular support for component testing and you may see it in as soon as next month!

**Reply by the-dietcoke-ofevil** [reply]
Angular support for component testing is here! ~ https://www.cypress.io/blog/2022/08/15/cypress-10-5-0-introducing-angular-component-testing/

**Comment by MikeShi42** [comment]
Hi everyone! After seeing a lot of the feedback and knowing how awesome Cypress Studio is in Cy v9, my team has spent some time putting together a Cypress plugin that replicates a lot of the original Cypress Studio experience for Cypress v10, so you can one-click start recording test steps right within your test browser.

You can check out our plugin repo here: https://github.com/DeploySentinel/cypress-recorder

This is just the initial release, so would love to hear any feedback on what we've built here!

![demo](https://github.com/DeploySentinel/cypress-recorder/raw/main/assets/demo.gif)

**Reply by victorcmarinho** [reply]
amazing, a very used feature that unfortunately has been removed, thank you very much for your contribution I will test it here

**Reply by MikeShi42** [reply]
thank you @victorcmarinho! Let me know what you think 😄

**Reply by alapanme** [reply]
Do you intend to keep this project open-sourced or at some point in the future, make this into a paid product?

**Reply by MikeShi42** [reply]
@alapanme we don't have any plans of turning the recorder plugin or extension into a paid product and intend on it staying open source (would love contributions!)

we'd love to give back to the community, and transparently, we justify the investment by hoping it helps build awareness for the other helpful tooling we create for teams/companies 😄

**Reply by plfgavilan** [reply]
This contribution is amazing. Looking forward to check it out!!
 I will give you my feedback!!

**Reply by SaraCooperQA** [reply]
@MikeShi42 Thanks for your contribution. I tried it out and it looks good for an initial release. Are you going to include the ability to right click and add Assertions like in Cypress Studio such as have text, have class, have attr, be visible etc?

![image](https://user-images.githubusercontent.com/106568626/181398064-141cda86-76a9-464c-957e-7318c53f3b63.png)

Also, In Chrome and Firefox when I right click I see Record hover over element, but I do not see Assert/wait for selected text.

Thanks

**Reply by MikeShi42** [reply]
Hi @SaraCooperQA that's awesome to hear. Right now to record assert/wait for required text, you'll want to highlight the text content you want to assert on before right clicking it (I know it's unfortunately a little confusing right now). Then the assert/wait for selected text assertion should open up.

As for being able to use different assertions, we want to open up more selector flexibility as opposed to just the one we pick out of the box (we do try to stick by best practices!), but I don't have a firm ETA on that. Though if you (or anyone else) has [some spare cycles and would like to put in some contributions](https://github.com/DeploySentinel/Recorder), we'd really appreciate it!

**Reply by donwelsh** [reply]
@MikeShi42 Unfortunately I don't have the coding skill to contribute, but always happy to test and write documentation if needed.

**Reply by jonwest3** [reply]
Hi @MikeShi42. Yes, thanks for the contribution. I tried it and while it is better than the other solutions out there, it is still not as integrated, as easy to use, or as feature rich as Studio was. I am still hoping they bring Studio back into v10 soon.

**Comment by rgembalik** [comment]
To me:
- Studio was not cannot-live-without-it feature, but definitely sped-up initial test drafting. 
- Flow introduced with Chrome Dev Tools recorder and converter is not as comfortable to use (mainly due to the conversion step)
- Could chrome-dev-tools version be updated to improve support? Maybe. I think if it was possible for the cypress recorder to launch some sort of chromium instance ready for recording and automatically converted its output to cypress format, then I would be happier about the entire conversion.
- Can cypress function without a recorder? Yeah, although I might have a hard time convincing my team to use cypress again or to update to 10. Especially since recordings in chrome dev tools do support puppeteer

**Comment by Zujaj** [comment]
I was just about to start Cypress studio after adding the latest version and found out that it's gone! Please kindly restore it, it's really an awesome feature especially for automating manual QA steps. Hope the cypress team would bring it to life soon.

**Comment by lmiller1990** [comment]
Hello all. Thank you for your continued feedback on Studio – it’s clear to us that it’s an important tool in a variety of use cases and workflows. We’re excited to announce that we’ve reprioritized restoring Studio in Cypress 10 based on your feedback. I am working on this now. You can track progress and what needs to be done to use Studio in Cypress 10 here. [#23230](https://github.com/cypress-io/cypress/issues/23230).

Please note that we intend to bring back Studio exactly as it was in Cypress 9, so Studio will not work for Component Testing (on initial release). If you need Studio in Component Testing, please let us know by commenting on this thread. Your feedback will help us in future iterations.

**Comment by pstakoun** [comment]
Hey all. I'm very happy to let everyone know that we have reimplemented Cypress Studio into Cypress 10 as of version 10.7.0, which was released today! Please see our [blog post](https://www.cypress.io/blog/2022/08/30/how-to-use-studio-in-cypress-10-7) for more info.

Thank you for your feedback. The best way for us to prioritize what we work on is by taking into account what our users have to say. I encourage you to direct additional feedback to our [Beta Feedback discussion](https://on.cypress.io/studio-beta), as we will be tracking that thread now that Studio is back in v10.

**Reply by plfgavilan** [reply]
great news!! Thank you very much!
